site/app/controllers/AuthenticationController.php

```diff
@@ -308,4 +309,185 @@ public function userSelection() {
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if the password is greater than or equal to 12 characters, and has the required characters
+     * @param string $password
+     */
+    public function isGoodPassword($password): bool {
+        return strlen($password) >= 12 && $this->checkChars($password);
+    }
+
+    /**
+     * Checks if the email extension is in the accepted emails JSON file
+     * @param string $email
+     */
+    public function isAcceptedEmail($email): bool {
+        $json = file_get_contents('/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/accepted_emails.json');
+        // Check if the file was read successfully
+        try {
+            $email_extension = explode('@', $email)[1];
+        }
+        catch (\Error $error) {
+            return false;
+        }
+
+        if ($json === false) {
+            return false;
+        }
+
+        // Decode the JSON file
+        $json_data = json_decode($json, true);
+
+        // Check if the JSON was decoded successfully
+        if ($json_data === null) {
+            return false;
+        }
+
+        return in_array($email_extension, array_keys($json_data), true);
+    }
+
+    /**
+     * Handles the submission of the new account creation form
+     */
+    #[Route("/authentication/self_add_user")]
+    public function addNewUser(): RedirectResponse {
+        // Check if the user is already logged in, if yes, redirect to home or another appropriate page
+        if ($this->logged_in) {
+            return new RedirectResponse($this->core->buildUrl(['home']));
+        }
+
+        // Should never happen, however they can visit this URL manually, so this is to prevent unwanted account creation.
+        if (!$this->core->getConfig()->isUserCreateAccount()) {
+            $this->core->addErrorMessage('Users cannot create their own account, Please have your system administrator add you.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'login']));
+        }
+
+        $user_id = $_POST['user_id'];
+        $email = $_POST['email'];
+        $password = $_POST['password'];
+        $confirm_password = $_POST['confirm_password'];
+
+        $queried_list = $this->core->getQueries()->getFullEmailList();
+
+        if (in_array($email, array_column($queried_list, 'user_email'), true)) {
+            $this->core->addErrorMessage('Email already exists');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if (in_array($user_id, array_column($queried_list, 'user_id'), true)) {
+            $this->core->addErrorMessage('User ID already exists');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if (!$this->isGoodPassword($password)) {
+            $this->core->addErrorMessage('Password does not meet the requirements.');
```

Comment:
![스크린샷 2024-07-30 오후 1 33 11](https://github.com/user-attachments/assets/48a7f5c2-edd6-44b5-9055-37879b001d16)

I think during the meeting, we decided to specify what the requirements are. Right now, I cannot tell what the reqs are.

Reply:
I am working on adding them to the form page